### PR TITLE
GUI getControlByName

### PR DESCRIPTION
**Pull request by msDestiny14**
Adding @PirateJC for ux check. https://github.com/BabylonJS/ThePirateCove/issues/93

### CODE REVIEW

**Review by RaananW — COMMENTED** [review]
Array find function does exactly what you want without creating unneeded arrays:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

It is not available in IE11, but as we are slowly removing support for this wonderful browser, I don't see a reason to support this browser with new functionality. Besides, it is already being used:

```javascript
private getControlByKey(key: string, value: string) : Nullable<Control> {
  return this._rootContainer.getDescendants().find(control => control[key] === value) || null;
}
```

**Comment by msDestiny14** [comment]
like this?

**Comment by msDestiny14** [comment]
wait actually what is @PirateJC and @PatrickRyanMS thoughts? While by key would be better technically is it too much to have users type an extra parameter?

**Comment by PatrickRyanMS** [comment]
I'm not sure I follow the question. Are we needing to pass a key value instead of the name of the control? It would be a more straight path for the user to be able to get by the name they gave the control instead of an ID or key value that we assign since the user will have to find that value to pass. Is that what we are suggesting here?

File: gui/src/2D/advancedDynamicTexture.ts [hunk 1]
```diff
@@ -272,6 +272,29 @@ export class AdvancedDynamicTexture extends DynamicTexture {
...
+    /**
+    * Will return the first control with the given id
+    * @param uniqueId defines the id to search for
+    * @return the first control found or null
+    */
+    public getControlById(uniqueId: number): Nullable<Control> {
+        return this.getControlByKey("uniqueId", uniqueId);
+    }
+
+    private getControlByKey(key: string, value: any): Nullable<Control> {
```
Comment (RaananW) [inline gui/src/2D/advancedDynamicTexture.ts:294]:
```suggestion
    private _getControlByKey(key: string, value: any): Nullable<Control> {
```

File: gui/src/2D/advancedDynamicTexture.ts [hunk 2]
```diff
@@ -272,6 +272,29 @@ export class AdvancedDynamicTexture extends DynamicTexture {
...
+        return this.getControlByKey("name", name);
+    }
+
+    /**
+    * Will return the first control with the given id
+    * @param uniqueId defines the id to search for
+    * @return the first control found or null
+    */
+    public getControlById(uniqueId: number): Nullable<Control> {
+        return this.getControlByKey("uniqueId", uniqueId);
```
Comment (RaananW) [inline gui/src/2D/advancedDynamicTexture.ts:291]:
```suggestion
        return this._getControlByKey("uniqueId", uniqueId);
```

File: gui/src/2D/advancedDynamicTexture.ts [hunk 3]
```diff
@@ -272,6 +272,29 @@ export class AdvancedDynamicTexture extends DynamicTexture {
...
         return this._rootContainer.getDescendants(directDescendantsOnly, predicate);
     }
+
+    /**
+    * Will return the first control with the given name
+    * @param name defines the name to search for
+    * @return the first control found or null
+    */
+    public getControlByName(name: string): Nullable<Control> {
+        return this.getControlByKey("name", name);
```
Comment (RaananW) [inline gui/src/2D/advancedDynamicTexture.ts:282]:
```suggestion
        return this._getControlByKey("name", name);
```

**Review by RaananW — APPROVED** [review]

**Pull request merged**